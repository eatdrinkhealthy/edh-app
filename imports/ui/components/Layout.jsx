// @flow
import React from "react";

type IProps = {
  children: React.Element<*>,
};

const Layout = (props: IProps) => {
  const { children, ...otherProps } = props;
  const newChild = React.cloneElement(children, otherProps);

  return (
    <div>
      {newChild}
    </div>
  );
};

export default Layout;

