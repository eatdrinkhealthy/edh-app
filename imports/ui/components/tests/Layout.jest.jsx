/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */

import React from "react";
import renderer from "react-test-renderer";
import Layout from "../Layout";

describe("<Layout />", function () {
  it("matches render snapshot", function () {
    const tree = renderer.create(<Layout><p>children</p></Layout>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
