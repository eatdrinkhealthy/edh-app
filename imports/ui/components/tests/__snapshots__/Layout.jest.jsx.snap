exports[`<Layout /> matches render snapshot 1`] = `
<div>
  <p>
    children
  </p>
</div>
`;
